app/javascript/controllers/admin/asset_controller.js

```diff
@@ -0,0 +1,50 @@
+import { Controller } from 'stimulus'
+
+export default class extends Controller {
+  static targets = ['spamButton', 'deleteButton']
+
+  initialize() {
+    if (this.data.get('deleted') == "true" ) {
+      this.hideDeleteButtonAndSpamButton()
+    } else if (this.data.get('spam') == "true") {
+      this.hideSpamButton()
+      this.showDeleteButton()
+    } else {
+      this.showSpamButton()
+      this.showDeleteButton()
+    }
+  }
+
+  connect() {
+    // console.log("Hello, Stimulus!", this.element)
+  }
+
+  //////////////////////////////////
+  // SPAM button
+  //////////////////////////////////
```

Comment:
We'll have to remove these comment blocks to please our current linter settings. If you don't want to use eslint locally I can see if code climate will run eslint. Re: these particular comments,  "SpamButton" is clear in the name of the methods, so I think we can toss em! Generally, I love comments, but prefer to reserve for the "why" or rationale of otherwise confusing things.